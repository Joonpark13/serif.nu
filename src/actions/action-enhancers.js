import { currentTermSelector } from 'selectors';

export const ENHANCE_WITH_CURRENT_TERM_ID = Symbol('ENHANCE WITH CURRENT TERM');
const termIdEnhancer = {
  id: ENHANCE_WITH_CURRENT_TERM_ID,
  mapState: state => ({ currentTermId: currentTermSelector(state).get('id') }),
};

const enhancers = [termIdEnhancer];
export default enhancers;
