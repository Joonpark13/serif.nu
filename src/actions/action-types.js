export const GET_SEARCH_RESULTS_REQUEST = 'GET_SEARCH_RESULTS_REQUEST';
export const GET_SEARCH_RESULTS_SUCCESS = 'SEARCH_RESULTS_SUCCESS';
export const GET_SEARCH_RESULTS_FAILURE = 'SEARCH_RESULTS_FAILURE';
export const CLEAR_SEARCH_RESULTS = 'CLEAR_SEARCH_RESULTS';
export const GET_SECTIONS_REQUEST = 'GET_SECTIONS_REQUEST';
export const GET_SECTIONS_SUCCESS = 'GET_SECTIONS_SUCCESS';
export const GET_SECTIONS_FAILURE = 'GET_SECTIONS_FAILURE';
export const SET_CURRENT_COURSE_NAME = 'SET_CURRENT_COURSE_NAME';
export const VIEW_SEARCH = 'VIEW_SEARCH';
export const ADD_SECTION = 'ADD_SECTION';
export const REMOVE_SECTION = 'REMOVE_SECTION';
export const GET_SCHOOLS_REQUEST = 'GET_SCHOOLS_REQUEST';
export const GET_SCHOOLS_SUCCESS = 'GET_SCHOOLS_SUCCESS';
export const GET_SCHOOLS_FAILURE = 'GET_SCHOOLS_FAILURE';
export const UPDATE_SEARCH_INPUT = 'UPDATE_SEARCH_INPUT';
export const FETCH_SEARCH_INDEX = 'FETCH_SEARCH_INDEX';
export const FETCH_SEARCH_INDEX_SUCCESS = 'FETCH_SEARCH_INDEX_SUCCESS';
export const FETCH_SEARCH_INDEX_FAILURE = 'FETCH_SEARCH_INDEX_FAILURE';
export const FETCH_SUBJECTS_REQUEST = 'FETCH_SUBJECTS_REQUEST';
export const FETCH_SUBJECTS_SUCCESS = 'FETCH_SUBJECTS_SUCCESS';
export const FETCH_SUBJECTS_FAILURE = 'FETCH_SUBJECTS_FAILURE';
