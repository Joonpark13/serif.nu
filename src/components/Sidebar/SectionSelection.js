import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import Divider from '@material-ui/core/Divider';

export const styles = {
  sectionsRoot: {
    padding: 16,
  },
  title: {
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  divider: {
    marginTop: 5,
  },
  sectionTitle: {
    fontWeight: 'bold',
  },
};

function SectionSelection({
  currentCourseName,
  sections,
  back,
  classes,
}) {
  return (
    <div className={classes.sectionsRoot}>
      <Typography variant="headline" className={classes.title}>
        {currentCourseName}
        <Button onClick={back}>Back</Button>
      </Typography>
      <Divider className={classes.divider} />
      <List>
        {sections.map(section => (
          <ListItem
            key={section.id}
            button
          >
            <ListItemText>
              <Typography variant="title" className={classes.sectionTitle}>{`Section ${section.section}`}</Typography>
              <Typography>{`${section.class_mtg_info[0].meet_t}`}</Typography>
              <Typography>{`${section.class_mtg_info[0].meet_l}`}</Typography>
              <Typography>
                {section.instructor.map((teacher, idx) => (
                  `${teacher}${idx !== section.instructor.length - 1 ? ', ' : ''}`))
                }
              </Typography>
            </ListItemText>
          </ListItem>
        ))
        }
      </List>
    </div>
  );
}

SectionSelection.propTypes = {
  currentCourseName: PropTypes.string.isRequired,
  sections: PropTypes.arrayOf(PropTypes.object).isRequired,
  back: PropTypes.func.isRequired,
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
};

export { SectionSelection as UnstyledSectionSelection };
export default withStyles(styles)(SectionSelection);
