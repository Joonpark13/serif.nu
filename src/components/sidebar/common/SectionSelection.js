import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/styles';
import { List } from '@material-ui/core';
import SectionResult from './SectionResult';
import SidebarHeader from '../SidebarHeader';

const useStyles = makeStyles({
  sectionsRoot: {
    padding: 16,
  },
  title: {
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  divider: {
    marginTop: 5,
  },
});

export default function SectionSelection({
  currentCourseName,
  sections,
  scheduledSections,
  addSection,
  back,
}) {
  const classes = useStyles();
  return (
    <div className={classes.sectionsRoot}>
      <SidebarHeader title={currentCourseName} back={back} />
      <List>
        {sections.map((section) => {
          const sectionAlreadyAdded = !!scheduledSections && scheduledSections.some(
            scheduledSection => scheduledSection.id === section.id,
          );
          return (
            <SectionResult
              key={section.id}
              section={section}
              disabled={sectionAlreadyAdded}
              addSection={addSection}
            />
          );
        })
        }
      </List>
    </div>
  );
}

SectionSelection.propTypes = {
  currentCourseName: PropTypes.string.isRequired,
  sections: PropTypes.arrayOf(PropTypes.object).isRequired,
  scheduledSections: PropTypes.arrayOf(PropTypes.object),
  addSection: PropTypes.func.isRequired,
  back: PropTypes.func.isRequired,
};

SectionSelection.defaultProps = {
  scheduledSections: [],
};
