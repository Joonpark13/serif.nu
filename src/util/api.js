/* eslint-disable import/prefer-default-export */
export function searchURL(query) {
  return `${process.env.API_URL}/search?term_id=4720&search_query=${query}`;
}

export function sectionsURL(termId, schoolAbbv, subjectAbbv, courseAbbv) {
  return `${process.env.API_URL}/sections?term_id=${termId}&school_abbv=${schoolAbbv}&subject_abbv=${subjectAbbv}&course_abbv=${courseAbbv}`;
}
