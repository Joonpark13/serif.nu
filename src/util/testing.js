/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import * as reactRedux from 'react-redux';
import { shallow } from 'enzyme';
import * as useSelector from './use-selector';

jest.mock('./use-selector');

// Helper function that returns a function which creates enzyme wrappers with default props
export function wrapperCreator(Component, defaultProps = {}) {
  let classes;
  return props => shallow(<Component classes={classes} {...defaultProps} {...props} />);
}

/* istanbul ignore next */
export function mockUseSelector(...args) {
  useSelector.default.mockReset();
  args.forEach(arg => useSelector.default.mockReturnValueOnce(arg));

  return useSelector.default;
}

/* istanbul ignore next */
export function mockUseDispatch() {
  const dispatchSpy = jest.fn();
  reactRedux.useDispatch.mockReturnValue(dispatchSpy);
  return dispatchSpy;
}
