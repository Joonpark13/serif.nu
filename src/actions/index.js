import * as actionTypes from './action-types';

export const clearSearchResults = () => ({
  type: actionTypes.CLEAR_SEARCH_RESULTS,
});

export function fetchSectionsForSearchRequest(schoolId, subjectId, courseId) {
  return {
    type: actionTypes.FETCH_SECTIONS_FOR_SEARCH_REQUEST,
    schoolId,
    subjectId,
    courseId,
  };
}

export function fetchSectionsForSearchSuccess(sections) {
  return {
    type: actionTypes.FETCH_SECTIONS_FOR_SEARCH_SUCCESS,
    sections,
  };
}

export function fetchSectionsForSearchFailure() {
  return {
    type: actionTypes.FETCH_SECTIONS_FOR_SEARCH_FAILURE,
  };
}

export const setCurrentCourseName = courseName => ({
  type: actionTypes.SET_CURRENT_COURSE_NAME,
  courseName,
});

export function viewSearch() {
  return {
    type: actionTypes.VIEW_SEARCH,
  };
}

export function addSectionFromSearch(section) {
  return {
    type: actionTypes.ADD_SECTION_FROM_SEARCH,
    section,
  };
}

export function addSectionWithAssociatedClassFromSearch(associatedClass) {
  return {
    type: actionTypes.ADD_SECTION_WITH_ASSOCIATED_CLASS_FROM_SEARCH,
    associatedClass,
  };
}

export const removeSection = (sectionId, sectionColor) => ({
  type: actionTypes.REMOVE_SECTION,
  sectionId,
  sectionColor,
});

export const removeAllClasses = () => ({
  type: actionTypes.REMOVE_ALL_CLASSES,
});

export function fetchSchoolsRequest() {
  return {
    type: actionTypes.FETCH_SCHOOLS_REQUEST,
  };
}

export function fetchSchoolsSuccess(schools) {
  return {
    type: actionTypes.FETCH_SCHOOLS_SUCCESS,
    schools,
  };
}

export function fetchSchoolsFailure() {
  return {
    type: actionTypes.FETCH_SCHOOLS_FAILURE,
  };
}

export function fetchSearchResultsRequest(searchInput) {
  return {
    type: actionTypes.FETCH_SEARCH_RESULTS_REQUEST,
    searchInput,
  };
}

export function fetchSearchResultsSuccess(searchResults) {
  return {
    type: actionTypes.FETCH_SEARCH_RESULTS_SUCCESS,
    searchResults,
  };
}

export function fetchSearchResultsFailure() {
  return {
    type: actionTypes.FETCH_SEARCH_RESULTS_FAILURE,
  };
}

export function updateSearchInput(searchInput) {
  return {
    type: actionTypes.UPDATE_SEARCH_INPUT,
    searchInput,
  };
}

export function fetchSubjectsRequest(schoolId) {
  return {
    type: actionTypes.FETCH_SUBJECTS_REQUEST,
    schoolId,
  };
}

export function fetchSubjectsSuccess(subjects) {
  return {
    type: actionTypes.FETCH_SUBJECTS_SUCCESS,
    subjects,
  };
}

export function fetchSubjectsFailure() {
  return {
    type: actionTypes.FETCH_SUBJECTS_FAILURE,
  };
}

export function fetchCoursesRequest(schoolId, subjectId) {
  return {
    type: actionTypes.FETCH_COURSES_REQUEST,
    schoolId,
    subjectId,
  };
}

export function fetchCoursesSuccess(courses) {
  return {
    type: actionTypes.FETCH_COURSES_SUCCESS,
    courses,
  };
}

export function fetchCoursesFailure() {
  return {
    type: actionTypes.FETCH_COURSES_FAILURE,
  };
}

export function fetchSectionsForBrowseRequest(schoolId, subjectId, courseId) {
  return {
    type: actionTypes.FETCH_SECTIONS_FOR_BROWSE_REQUEST,
    schoolId,
    subjectId,
    courseId,
  };
}

export function fetchSectionsForBrowseSuccess(sections) {
  return {
    type: actionTypes.FETCH_SECTIONS_FOR_BROWSE_SUCCESS,
    sections,
  };
}

export function fetchSectionsForBrowseFailure() {
  return {
    type: actionTypes.FETCH_SECTIONS_FOR_BROWSE_FAILURE,
  };
}


export function changeBrowseLevel(browseLevel) {
  return {
    type: actionTypes.CHANGE_BROWSE_LEVEL,
    browseLevel,
  };
}

export function viewSectionSelection() {
  return {
    type: actionTypes.VIEW_SECTION_SELECTION,
  };
}

export function selectSchoolInBrowse(schoolId) {
  return {
    type: actionTypes.SELECT_SCHOOL_IN_BROWSE,
    schoolId,
  };
}

export function selectSubjectInBrowse(subjectId) {
  return {
    type: actionTypes.SELECT_SUBJECT_IN_BROWSE,
    subjectId,
  };
}

export function selectCourseInBrowse(courseId) {
  return {
    type: actionTypes.SELECT_COURSE_IN_BROWSE,
    courseId,
  };
}

export function addSectionFromBrowse(section) {
  return {
    type: actionTypes.ADD_SECTION_FROM_BROWSE,
    section,
  };
}

export function addSectionWithAssociatedClassFromBrowse(associatedClass) {
  return {
    type: actionTypes.ADD_SECTION_WITH_ASSOCIATED_CLASS_FROM_BROWSE,
    associatedClass,
  };
}

export function sectionHover(section) {
  return {
    type: actionTypes.SECTION_HOVER,
    section,
  };
}

export function sectionHoverOff() {
  return {
    type: actionTypes.SECTION_HOVER_OFF,
  };
}

export function associatedClassHover(associatedClass) {
  return {
    type: actionTypes.ASSOCIATED_CLASS_HOVER,
    associatedClass,
  };
}

export function associatedClassHoverOff() {
  return {
    type: actionTypes.ASSOCIATED_CLASS_HOVER_OFF,
  };
}
