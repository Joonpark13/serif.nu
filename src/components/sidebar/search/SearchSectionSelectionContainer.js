import React from 'react';
import { useDispatch } from 'react-redux';
import useSelector from 'util/use-selector';
import { currentCourseNameSelector, currentSectionsSelector, sectionsSelector } from 'selectors';
import { viewSearch, addSectionFromSearch } from 'actions';
import SectionSelection from '../common/SectionSelection';

function SearchSectionSelectionContainer() {
  const currentCourseName = useSelector(currentCourseNameSelector);
  const sections = useSelector(currentSectionsSelector);
  const scheduledSections = useSelector(sectionsSelector);
  const dispatch = useDispatch();

  return (
    <SectionSelection
      currentCourseName={currentCourseName}
      sections={sections}
      scheduledSections={scheduledSections}
      back={() => dispatch(viewSearch())}
      addSection={section => dispatch(addSectionFromSearch(section))}
    />
  );
}

export default SearchSectionSelectionContainer;
