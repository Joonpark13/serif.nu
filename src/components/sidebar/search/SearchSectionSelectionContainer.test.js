import { viewSearch, addSectionFromSearch } from 'actions';
import { mockUseSelector, wrapperCreator, mockUseDispatch } from 'util/testing';
import SectionSelection from '../common/SectionSelection';
import SearchSectionSelectionContainer from './SearchSectionSelectionContainer';

describe('SearchSectionSelectionContainer', () => {
  const getWrapper = wrapperCreator(SearchSectionSelectionContainer);
  let dispatchMock;

  beforeEach(() => {
    mockUseSelector('course name', [], []);
    dispatchMock = mockUseDispatch();
  });

  it('renders correctly', () => {
    const wrapper = getWrapper();

    expect(wrapper.get(0)).toMatchSnapshot();
  });

  it('navigates to search when navigating back', () => {
    const wrapper = getWrapper();
    wrapper.find(SectionSelection).props().back();

    expect(dispatchMock).toHaveBeenCalledWith(viewSearch());
  });

  it('adds section', () => {
    const wrapper = getWrapper();
    const section = {};
    wrapper.find(SectionSelection).props().addSection(section);

    expect(dispatchMock).toHaveBeenCalledWith(
      addSectionFromSearch(section),
    );
  });
});
