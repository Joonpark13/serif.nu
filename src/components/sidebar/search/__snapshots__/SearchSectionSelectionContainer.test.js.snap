// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchSectionSelectionContainer renders correctly 1`] = `
<SectionSelection
  addSection={[Function]}
  back={[Function]}
  currentCourseName="course name"
  scheduledSections={Array []}
  sections={Array []}
/>
`;
